README.md

```diff
@@ -3,7 +3,78 @@ Legends of the Wulin Character Sheet
 
 <p align="center">
     <a href="https://circleci.com/docs/1.0/status-badges/">
-        <img src="https://circleci.com/gh/gentilboulet/LotWCS.svg?&style=shield&circle-token=784ac32effdcc6931968ae398ce0c87289181e68"
+        <img src="https://circleci.com/gh/gentilboulet/LotWCS.svg?&style=shield"
              alt="build status">
     </a>
 </p>
+
+This app is designed to help players & wulin sage alike to create character sheets for the role-playing game Legends of the Wulin.
+
+Currently, there are no plans to provide a character database to users.
+Instead, LotWCS will provide an import/export feature for you to store your characters.
+
+# Technical stuff
+Here we will bore you with technical considerations, for developers of LotWCS mainly.
+
+## App Model
+This application uses the standard React-Redux pattern.
+The whole application relies on a Redux Store to store the character information.
+The character evolves from an initial state to the current state through user actions.
+
+The Redux architecture handle state evolutions by applying a list of actions to the current state, returning a new updated state.
+The Redux Store handle the dispatching for actions (from user inputs mainly) to the reducers.
+
+The state is used to feed React Components to display the character.
+React Components are subdivided by responsibility: container Components are there to link the store to render Components.
+
+## File hierarchy
+The src directory is divided by responsibility:
+
+### actions
+Here are the action creators, a lot of helper functions present to generate an input to a specific action understandable by the reducers.
+
+### components
+__This directory regroups "renderer" components, handling only rendering and catching user inputs. __
+
+Theirs props must provide everything they need to know to do their job.
+
+### constants
+Constants definitions, mainly used to specify available action.type value, on which the reducers rely to identify actions.
+
+### containers
+__This directory regroups "containers" components and other functions for reading elements from the state. __
+
+Those are obtained by connecting the store and the state to a "renderer" component.
+To do so, three steps are taken for each one:
+* mapStateToProps: maps the state to the props, by transforming it, by adding information from static data ;
```

Comment:
refer to the [official documentation](https://github.com/reactjs/react-redux/blob/9961062afe35a9147b0ea819e7087170e66cd5f5/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)